marlowe-playground-client/src/Contrib/Halogen/State/Record.purs

```diff
@@ -0,0 +1,74 @@
+module Contrib.Halogen.State.Record where
+
+import Contrib.Type.Data.Symbol.Transform (class ToUpperFirst) as Symbol.Transform
+import Heterogeneous.Folding (class FoldingWithIndex, class HFoldlWithIndex, hfoldlWithIndex)
+import Prim.Row (class Cons, class Lacks) as Row
+import Prim.RowList (class RowToList) as RL
+import Prim.Symbol (class Append) as Symbol
+import Record (insert, modify, set) as Record
+import Type.Prelude (class IsSymbol, Proxy(..))
+
+-- | This is not strictly halogen specific folding...
+-- |
+-- | Given a proxy to the record (like "state record")
+-- | we produce a dispatcher (another record).
+-- |
+-- | This dispatcher when used with `Variant.match`
+-- | accepts a `Variant` value (an "action") and dispatches it
+-- | using its label in a type safe manner over our state value.
+data SettersDispatchStep
+  = SettersDispatchStep
+
+-- | We use `FoldingWithIndex` here and not
+-- | `MappingWithIndex` because mapping requires
+-- | a value to operate on and folding allows us to build up
+-- | a value just from the `Proxy (RowList k)`.
+instance purtyProblem0 ::
```

Comment:
what is this instance name abount?

Reply:
I'm leaving instance names in the code because `purty` enforces it. We can drop this stuff naming instances boilerplate when we move to `purs-tidy` because instance names are not required any more.